cylc/flow/platforms.py

```diff
@@ -417,30 +421,27 @@ def fail_if_platform_and_host_conflict(task_conf, task_name):
             )
 
 
-def get_install_target_from_platform(platform):
+def get_install_target_from_platform(platform: Dict[str, Any]) -> str:
     """Sets install target to configured or default platform name.
```

Comment:
```suggestion
Get install target if set on platform, else return platform name.
```

Reply:
But it has the side effect of mutating the platform passed to it. I wonder if that should be moved to somewhere else. I also wonder if we should create a class to represent a Platform, rather than relying on dicts

Reply:
+1 for Class for platform from me!

Reply:
It seems like overkill to me, but it's possible that the install targets has added complexity.
I suppose we could use a dataclass?